src/graph/rome_models.cc

```diff
@@ -1367,12 +1616,23 @@ ncclResult_t parseRome4P2H(struct ncclTopoSystem* system, struct ncclTopoGraph*
   }
   if (i < romeTopo.nGpus) match_nbio = false;
 
+  // check if NCCL_IB_GID_INDEX=3 -- needed for RoCE systems
+  const char* ncclIbGidIndex = ncclGetEnv("NCCL_IB_GID_INDEX");
+  bool isRoCE = 0;
+  if (ncclIbGidIndex) isRoCE = strcmp(ncclIbGidIndex, "3") == 0;
+
   for (i = 0; i < sizeof(romeTopoModels)/sizeof(romeTopoModels[0]); i++) {
     bool ignore_cpu = checkOption(romeTopoModels[i].options, "noCpuCheck");
     if (!ignore_cpu && (arch != NCCL_TOPO_CPU_ARCH_X86 || vendor != NCCL_TOPO_CPU_VENDOR_AMD || model != NCCL_TOPO_CPU_TYPE_ROME))
       continue;
+
     bool ignore_numa = checkOption(romeTopoModels[i].options, "disableNumaMatching");
     if (!ignore_numa && romeTopo.nCpus != romeTopoModels[i].nCpus) continue;
+
+    // check if isRoCE=1 is defined in model struct
+    bool optionsIsRoCE = checkOption(romeTopoModels[i].options, "isRoCE");
```

Comment:
what happen with models doesn't have "isRoCE" option?

Reply:
`checkOption` returns false if "isRoCE" is not found in options, so `optionsIsRoCE` will be 0.\
variable `isRoCE` in line 1621 is initialized as false, so line 1634: `if (optionsIsRoCE != isRoCE) continue;` will not come into effect.\

verified with model match for single node run, which shows "Rome model index 38 with GPU mapping: 0 1 2 3 4 5 6 7" i.e. `rome_model_79` which does not have isRoCE in options

Reply:
if the purpose is to skip "if (optionsIsRoCE != isRoCE) continue" in this case, shoulsn't it be:
if (!optionsIsRoCE && optionsIsRoCE != isRoCE) continue